src/dbnode/storage/shard.go

```diff
@@ -2117,6 +2127,31 @@ func (s *dbShard) cacheShardIndices() error {
 	return nil
 }
 
+func (s *dbShard) closeShardIndex() error {
+	retrieverMgr := s.opts.DatabaseBlockRetrieverManager()
+	// May be nil depending on the caching policy.
+	if retrieverMgr == nil {
+		return nil
+	}
+
+	retriever, err := retrieverMgr.Retriever(s.namespace)
+	if err != nil {
+		return err
+	}
+
+	s.logger.Debug("closing shard indices", zap.Uint32("shard", s.ID()))
+	if err := retriever.CloseShardIndices([]uint32{s.ID()}); err != nil {
+		s.logger.Error("closing shard indices error",
```

Comment:
By preference, I might wrap the error instead using `https://github.com/m3db/m3/tree/master/src/x/errors.Wrap` instead of logging it here--typically speaking my guideline is to log errors at the highest level possible in the call stack, lest you get things like:

```
error at the lowest level!
error at the next lowest level!
...
request failed!
```
all the way up the stack.

Btw just as a general note--that package is basically a more performant version of https://github.com/pkg/errors, which itself is starting to be deprecated in favor of %w in Errorf--https://blog.golang.org/go1.13-errors

Reply:
I agree with the pattern you suggested. It avoid duplicate logging. This error is already being logged here - https://github.com/m3db/m3/blob/542d128b38678dcc83c10169ecdedd2ef63d9cbc/src/dbnode/storage/namespace.go#L511

Made the change.